CastleBoy/entity.cpp

```diff
@@ -968,7 +972,7 @@ bool Entities::moveCollide(int16_t x, int8_t y, int8_t offsetX, int8_t offsetY,
             // moving platform collide only if player is going down on it
             // and player is right on platform
             collide = true;
-            if (ab.everyXFrames(3))
+            if (!Map::collide(x, y, hitbox))
```

Comment:
This is checking the old position of the player which is wrong. I think it should be:

```
if(ab.everyXFrames(3))
{
  int16_t newPlayerX = Player::pos.x + getMovingPlatformDirection(entity);
  if (!Map::collide(newPlayerX, y, hitbox))
  {
    Player::pos.x = newPlayerX;
  }
}
```